webapp/src/components/Analysis/UtterancesTable.tsx

```diff
@@ -390,6 +414,89 @@ const UtterancesTable: React.FC<Props> = ({
   ];
 
   const searchString = constructSearchString(pipeline);
+
+  const removeSpecialCharacters = (text: string) =>
+    text.replace(/[^a-zA-Z0-9_]/g, "");
+
+  const parseCSVData = (data: string) => {
+    const headers: string[] = data.slice(0, data.indexOf("\n")).split(",");
+    const headersWithoutSpecialCharacters: string[] = headers.map((h) =>
+      removeSpecialCharacters(h)
+    ); // This line is to remove all the extra space and special characters if any except underscore, alphabets and numbers.
+    const validHeaders = ["row_idx", "proposed_action"].map((header) =>
+      headersWithoutSpecialCharacters.includes(header)
+    );
+    if (validHeaders.some((valid) => !valid)) {
+      showModalMessage(
+        "The CSV file did not have the row_idx and proposed_action column headers to update the proposed action."
+      );
+      showModal(true);
+    } else {
+      const parsedData: UtterancePatch[] = data
+        .slice(data.indexOf("\n") + 1)
+        .split("\n")
+        .map((row) => {
+          if (row) {
+            const values = row
+              .split(",")
+              .map((r) => removeSpecialCharacters(r));
+            const obj = headersWithoutSpecialCharacters.reduce(
+              (object, header, index) => {
+                return Object.assign(object, {
+                  [API_HEADERS[header as keyof typeof API_HEADERS]]:
+                    removeSpecialCharacters(values[index]),
+                });
+              },
+              {}
+            );
+            return obj as UtterancePatch;
+          } else {
+            showModalMessage("There are no records in the CSV file.");
+            showModal(true);
+            return {} as UtterancePatch;
+          }
```

Comment:
Thanks to the `.trim()` I am suggesting, there shouldn't be an empty row anymore, so no need for this `if`.
Also, I think parsing the row can be simplified to something like this:
```suggestion
            const [persistentId, dataAction] = row.split(",");
            return { persistentId, dataAction };
```